src/library/ButtonGroup/ButtonGroup.js

```diff
@@ -45,17 +45,42 @@ type State = {
 export const componentTheme = (baseTheme: Object) => ({
   ButtonGroupButton_backgroundColor_checkedDisabled: baseTheme.color_gray_40,
   ButtonGroupButton_border_disabled: `solid 1px ${baseTheme.borderColor}`,
+  ButtonGroupButton_borderColor_active: baseTheme.borderColor_theme_active,
+  ButtonGroupButton_borderColor_hover: baseTheme.borderColor_theme_hover,
   ButtonGroupButton_borderStartColor: baseTheme.borderColor,
   ButtonGroupButton_borderStartColor_checked: 'currentcolor',
   ButtonGroupButton_color_checkedDisabled: baseTheme.color_gray_60,
 
   ...baseTheme
 });
 
-const styles = ({ fullWidth, theme: baseTheme }) => {
+const styles = ({ fullWidth, theme: baseTheme, variant }) => {
   let theme = componentTheme(baseTheme);
+  if (variant) {
+    // prettier-ignore
+    theme = {
+      ...theme,
+      ButtonGroupButton_borderColor_active: theme[`borderColor_${variant}_active`],
+      ButtonGroupButton_borderColor_hover: theme[`borderColor_${variant}_hover`]
+    };
+  }
   const { direction } = theme;
   const rtl = direction === 'rtl';
+  const borderStartColorProperty = rtl ? 'borderRightColor' : 'borderLeftColor';
```

Comment:
For these rtl-dependent properties, I think some better names might be `startBorderColorProperty`, `endBorderColorProperty`, `startBorderBottomRadiusProperty`, `startBorderTopRadiusProperty`, `endBorderBottomRadiusProperty`, and`endBorderTopRadiusProperty`. Mostly because the properties themselves are `borderColor` and `border<Corner>Radius`

Reply:
Yeah, I was struggling between the two options; I thought this way made sense because it ends up being `border<direction>Color` so I was replacing `<direction>` with `start` or `end`; but I think your way makes sense too & I trust your judgement! :)